test/python/requirements-test.txt

```diff
@@ -0,0 +1,2 @@
+pytest~=6.2.5
```

Comment:
`pytest~=7.0.1` would still support python3.6. Maybe update the version?

Reply:
I actually learned from the SyReC PR, that what we are doing in the Python tests isn't actually pytest at all (it's Python's own `unittest`).
So the real question is, whether we do need this here at all 🙃

Reply:
I decided to keep it and change the tests to the `pytest` structure. This seems to incur much less boilerplate code.
Also I decided to drop support for Python 3.6 as is has reached its end of life a couple months ago.

Reply:
Looks good to me :)